apps/ledger-live-desktop/src/renderer/modals/SignMessage/Body.tsx

```diff
@@ -23,52 +32,43 @@ type OwnProps = {
   data: Data;
 };
 type Props = OwnProps;
-const steps: Array<St> = [
-  {
-    id: "summary",
-    label: <Trans i18nKey="signmessage.steps.summary.title" />,
-    component: StepSummary,
-    footer: StepSummaryFooter,
-  },
-  {
-    id: "sign",
-    label: <Trans i18nKey="signmessage.steps.sign.title" />,
-    component: StepSign,
-    onBack: ({ transitionTo }: StepProps) => {
-      transitionTo("summary");
-    },
-  },
-];
+
 const Body = ({ onClose, data }: Props) => {
-  const { t } = useTranslation();
-  const [stepId, setStepId] = useState("summary");
-  const handleStepChange = useCallback((e: { id: string }) => setStepId(e.id), [setStepId]);
-  const stepperOnClose = useCallback(() => {
-    if (onClose) {
-      onClose();
-    }
-    if (data.onClose) {
-      data.onClose();
-    }
-  }, [data, onClose]);
-  const stepperProps = {
-    title: t("signmessage.title"),
-    account: data.account,
-    onStepChange: handleStepChange,
-    stepId,
-    steps,
-    isACRE: data.isACRE,
-    useApp: data.useApp,
-    dependencies: data.dependencies,
-    message: data.message,
-    onConfirmationHandler: data.onConfirmationHandler,
-    onFailHandler: data.onFailHandler,
-    onClose: stepperOnClose,
-  };
+  const dispatch = useDispatch();
+  const deviceBlocked = useDeviceBlocked();
+  const request = useMemo(() => {
+    const appRequests = dependenciesToAppRequests(data.dependencies);
+    return {
+      account: data.account as Account,
+      message: data.message,
+      appName: data.useApp,
+      dependencies: appRequests,
+      isACRE: data.isACRE,
+    };
+  }, [data]);
   return (
-    <Stepper {...stepperProps}>
-      <Track onUnmount event="CloseModalWalletConnectPasteLink" />
-    </Stepper>
+    <ModalBody
+      onClose={deviceBlocked ? undefined : onClose}
```

Comment:
I'm not sure this logic is present in the parent modal

Reply:
You could keep this logic here @Antoine-bls83 but I think we might have wanted to update the parent component to also block the close if the device is in a certain state